python/ray/data/tests/test_execution_optimizer_limit_pushdown.py

```diff
@@ -34,46 +34,58 @@ def _check_valid_plan_and_result(
         assert op in ds.stats(), f"Operator {op} not found: {ds.stats()}"
 
 
-def test_limit_pushdown_conservative(ray_start_regular_shared_2_cpus):
-    """Test limit pushdown behavior - pushes through safe operations."""
-
-    def f1(x):
-        return x
-
-    def f2(x):
-        return x
-
-    # Test 1: Basic Limit -> Limit fusion (should still work)
+def test_limit_pushdown_basic_limit_fusion(ray_start_regular_shared_2_cpus):
```

Comment:
Except for the tests listed below, the Ray Data pipelines in this diff aren't guaranteed to produce rows in a specific output order. So, if you don't set `check_ordering=False`, the tests might raise false negatives.

Could you update the appropriate tests to use `check_ordering=False`?

List of tests with guaranteed order (okay to use  `check_ordering=True`):
* test_limit_pushdown_union_with_sort
* test_limit_pushdown_complex_interweaved_operations
* test_limit_pushdown_stops_at_sort